introduction_to_applying_machine_learning/deepar_chicago_traffic_violations/deepar_chicago_traffic_violations.ipynb

```diff
@@ -0,0 +1,467 @@
+    "\n",
+    "We use the [Automatic model tuning] (https://docs.aws.amazon.com/sagemaker/latest/dg/automatic-model-tuning.html) or Hyperparameter optimization for identifying the best values for the [DeepAR hyperparameters](https://docs.aws.amazon.com/sagemaker/latest/dg/deepar_hyperparameters.html). The Automatic model tuning job will kick of 10 parallel jobs (set by by max_parallel_jobs) to search the best hyperparameters for this dataset. The jobs will try to minimize the root mean square error on the test dataset using predicted and actual values.\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.tuner import IntegerParameter, CategoricalParameter, ContinuousParameter, HyperparameterTuner\n",
+    "from sagemaker.amazon.amazon_estimator import get_image_uri\n",
+    "\n",
+    "\n",
+    "container = get_image_uri(region_name=region,\n",
+    "                          repo_name='forecasting-deepar')\n",
+    "\n",
+    "deepar = sagemaker.estimator.Estimator(container,\n",
+    "                                       role,\n",
+    "                                       train_instance_count=1,\n",
+    "                                       train_instance_type='ml.m4.xlarge',\n",
+    "                                       train_use_spot_instances=True, # use spot instances\n",
+    "                                       train_max_run=3600, # max training time in seconds\n",
+    "                                       train_max_wait=3600, # seconds to wait for spot instance\n",
+    "                                       output_path='s3://{}/{}'.format(bucket, s3_output_path),\n",
+    "                                       sagemaker_session=sess)\n",
+    "freq = 'D'\n",
+    "context_length = 30\n",
+    "\n",
+    "deepar.set_hyperparameters(time_freq=freq,\n",
+    "                           context_length=str(context_length),\n",
+    "                           prediction_length=str(prediction_length))\n",
+    "\n",
+    "hyperparameter_ranges = {'mini_batch_size': IntegerParameter(100, 400),\n",
+    "                         'epochs': IntegerParameter(200, 400),\n",
+    "                         'num_cells': IntegerParameter(30,100),\n",
+    "                         'learning_rate': ContinuousParameter(0.0001, 0.1)}\n",
+    "\n",
+    "objective_metric_name = 'test:RMSE'\n",
+    "\n",
+    "tuner = HyperparameterTuner(deepar,\n",
+    "                            objective_metric_name,\n",
+    "                            hyperparameter_ranges,\n",
+    "                            max_jobs=10,\n",
+    "                            strategy='Bayesian',\n",
+    "                            objective_type='Minimize',\n",
+    "                            max_parallel_jobs=10,\n",
```

Comment:
This could be fine for a demo notebook, but with max_parallel_jobs = max_jobs = 10, there won't be much tuning done.

Reply:
agreed, added comments.